wnfs/src/private/directory.rs

```diff
@@ -47,15 +47,15 @@ pub type PrivatePathNodesResult = PathNodesResult<PrivateDirectory>;
 #[derive(Debug, Clone, PartialEq)]
 pub struct PrivateDirectory {
     pub header: PrivateNodeHeader,
-    pub content: PrivateDirectoryContent,
+    pub(crate) content: PrivateDirectoryContent,
 }
 
 #[derive(Debug)]
 pub struct PrivateDirectoryContent {
     persisted_as: OnceCell<Cid>,
-    pub previous: BTreeSet<(usize, Encrypted<Cid>)>,
-    pub metadata: Metadata,
-    pub entries: BTreeMap<String, PrivateRef>,
+    pub(crate) previous: BTreeSet<(usize, Encrypted<Cid>)>,
+    pub(crate) metadata: Metadata,
+    pub(crate) entries: BTreeMap<String, PrivateLink>,
```

Comment:
It seemed like it'd make sense to make the `PrivateDirectoryContent` and related things `pub(crate)`.
`PrivateLink` isn't in the public API, so that'd leak otherwise.
Also, we've got stuff like `.get_metadata()`, so I felt it'd be weird if you could both do `.metadata` and `.get_metadata()`.

I'm somewhat unsure about `.header`, though. We use it in the examples, but I'm thinking perhaps we should eventually figure out a good API for developers to get what they want to get from the `header`. Technically I'd consider it an implementation detail that there's a separate header for directories (and I can see ways in which we'll change the inner data structures in the future to separate them differently).